pkg/ast/prune.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2018 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package ast
+
+func (desc *Description) Prune(list []Pos) *Description {
```

Comment:
Make this accept map[ast.Node]bool, this will simplify (and speedup) the check below. And then convert slice to map in fidlgen.

Reply:
Changed to `Filter` that accepts an arbitrary predicate, so that it can be used when filtering based on file names.